### PR TITLE
Fix file uploading issues

### PR DESCRIPTION
**Pull request by zkraime**
<h2>Purpose</h2>
To fix issue #2822 and another issue I discovered which is that the user is unable to cancel in-progress uploads. 
<h2>Changes</h2>
The user can now 'x' out queued and in-progress uploads. 
<h2>To Do</h2>
Clicking 'x' on a file that exceeds 128mb removes the row. 
However, when the warning message disappears, the row reappears (it can be 'x'ed again). 
More perplexingly, trying to upload another file exceeding 128mb causes the previous row to reappear: 

screen shot 2015-07-06 at 3 56 27 pm

So, that has to be fixed. 
[edit] Should be fixed now

<h2>Notes</h2>

I noticed that in the cancelUploads function, getQueuedFiles excludes the in-progress upload and therefore returns 1 less than it should (or 0 if only one file is being uploaded - and so the for loop doesn’t iterate, resulting in the button’s apparent lack of response when clicked)

So, I used getActiveFiles instead since it includes files with states ‘queued’ OR ‘uploading’ (the files can have one state at a time it seems) 

This seems to fix the issue, except that when the user clicks ‘x’ on the ‘uploading’ file, it results in a TypeError because _fangornDropzoneError tries to access xhr.status when xhr is undefined. 
Then I wondered, why is the drop zone error hook being run? 

Looked into it, realized it happens in the first conditional of Dropzone’s removeFile which is called in fangorn.js cancelUploads function. 
screen shot 2015-07-06 at 4 10 46 pm
Because the in-progress upload has status 'uploading', it calls dropzone's cancelUpload which eventually triggers drozone's error hook. 

So, I figured an easy fix would be to change the status of that file to 'queued' before calling removeFile on it.

### CODE REVIEW

**Comment by zkraime** [comment]
@sloria 
Please give this a first pass, thank you!

File: website/static/js/fangorn.js [hunk 1]
```diff
@@ -115,10 +115,14 @@ function findByTempID(parent, tmpID) {
 
 function cancelUploads (row) {
     var tb = this;
-    var filesArr = tb.dropzone.getQueuedFiles();
+    var filesArr = tb.dropzone.getActiveFiles();
+    var rejectedFilesArr = tb.dropzone.getRejectedFiles();
     for (var i = 0; i < filesArr.length; i++) {
         var j = filesArr[i];
         if(!row){
+            if (j.status === 'uploading') {
```
Comment (sloria) [inline website/static/js/fangorn.js]: This seems suspect; `status` is set by Dropzone, correct? I'm not sure what the side effects of changing it here will be.

**Comment by sloria** [comment]
https://github.com/CenterForOpenScience/osf.io/pull/3381 is supposed to address #2822, so I'm unsure if this PR will also be necessary. @zkraime Can you take a look at that PR and report whether it fixes the issues you were addressing?

**Comment by zkraime** [comment]
@sloria Sure I'll go ahead and do that

File: website/static/js/fangorn.js (hunk 1 above)
Comment (zkraime) [inline website/static/js/fangorn.js]: @sloria Right. At first I wanted to create a status setter function in dropzone.js, but that involved the overhead of overriding dropzone. 
I couldn't think of a third solution

**Comment by zkraime** [comment]
@sloria There is no conflict between this and #3381 as far as I can tell

**Comment by zkraime** [comment]
Update: tested on staging and it looks as though the issue has been addressed

**Pull request closed without merging**